app/api/controllers/book_api.rb

```diff
@@ -0,0 +1,50 @@
+class BookApi < Grape::API
+  resource :books do
+    desc 'GET /api/v1/books'
+    get do
+      books = Book.all
+      BookSerializer.new(books).serializable_hash
+    end
+
+    desc 'POST /api/v1/books'
+    params do
+      requires :name, type: String
+      requires :release, type: Date
+      optional :description
+    end
+    post do
+      book = Book.create(declared(params).merge(user: current_user))
+      BookSerializer.new(book).serializable_hash
+    end
```

Comment:
1. Leave a line break to separate the last line (line that return result) to the above ones
2. use `render_json(:ok, <serialized-hash>)` for GET, POST, PUT
3. use `render_json(:no_content, {})` for DELETE

```suggestion
    post do
      book = Book.create(declared(params).merge(user: current_user))
      
      render_json(:ok, BookSerializer.new(book).serializable_hash)
    end
```

Reply:
To have `render_json`, you need to create "app/api/v1/helpers/response_helper.rb"

```
# frozen_string_literal: true

module V1
  module Helpers
    module ResponseHelper
      extend Grape::API::Helpers

      class NotFoundPlatformKeyError < StandardError; end

      def render_json(symbol, data = {})
        code = SYMBOL_TO_STATUS_CODE[symbol]
        status code
        error!(symbol) if code >= 400

        present data
      end
    end
  end
end
```